### DISCUSSION TITLE
Setting up staging and production workspaces

### DISCUSSION THREAD

**Discussion by theocampbell** [sanity-io/sanity#4226] [Sanity Studio Support]
Hi All,
I am having an issue setting up production and staging workspaces that I can't seam to find the solution for. 
I am using Sanity V 3.5.0 and have created two workspaces in sanity.config.js as per the documentation: 
```
export` default defineConfig([
  {
    name: 'CircuLaw',
    title: 'CircuLaw',
    projectId: '2vfoxb3h',
    dataset: 'production',
    basePath: '/studio',
    plugins: [
      deskTool({
        structure: Structure,
      }),
      visionTool(),
    ],
    schema: {
      types: schemaTypes,
    },
    studio: {
      components: {
        logo: Logo,
      },
    },
  },
  {
    name: 'CircuLawStaging',
    title: 'CircuLaw Staging',
    projectId: '2vfoxb3h',
    dataset: 'staging',
    basePath: '/staging',
    plugins: [
      deskTool({
        structure: Structure,
      }),
      visionTool(),
    ],
    schema: {
      types: schemaTypes,
    },
  },
`]);

```
The thing is when I add content into one workspace it also adds it into the other. For example I added content into the production studio and in the desk view of the staging studio I can see the same content and vice versa. What is weird is that when I use the vision tool in production I get the correct response with all the content but in the staging vision tool there is no content. 

Anyone else having this issue? Perhaps it is linked to sanity.cli.js? Mine is: 

```
export` default defineCliConfig({
  api: {
    projectId: '2vfoxb3h',
    dataset: 'production',
  },
});
```

I am also getting an error message as described in these bugs: https://github.com/sanity-io/sanity/issues/3992 and https://github.com/sanity-io/sanity/issues/2386

**Comment by ahmedghazi** [comment]
Try this
```
import {defineCliConfig} from 'sanity/cli'

export default defineCliConfig({
  api: {
    projectId: 'xxx',
    dataset: 'production',
  },
  graphql: [
    {
      id: 'production',
      workspace: 'production',
    },
    {
      id: 'preprod',
      workspace: 'preprod',
    },
  ],
})
```
